scapy/contrib/bgp_fields.py

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""BGP-4 disector: fields for implementing MPBGP extensions"""
+
+import socket
+if not socket.has_ipv6:
+    raise socket.error("can't use AF_INET6, IPv6 is disabled")
```

Comment:
Hi Pierre,

well yes, this is the next thing that I didn't find correctly documented
anywhere. My understanding is that we will only support AF_INET6 related
functionality if socket.has_ipv6 is True.

If socket.has_ipv6 is False, I doubt that we will be able to decode, let
alone generate, IPv6 updates.

Having arrived to this point, I'd also like to start discussing if we can
create a section in conf that is BGP-4 specific.

I have always struggled with older BGP-4 captures, that only supported
16-bit Autonomous System numbers. The mechanics of this would be:

We have a variable (e.g. conf.bgp.use_asn4) originally set to False.
In a program that implements the interaction with a real network device, we
would set this to true if we want to use 4-byte ASNs and the capability
negotiation led to discover that the peer supports it.
In a program that digests captures, this could be set by the program (CLI
arguments?)

/PA



On 9 February 2016 at 08:29, Pierre Lalet <notifications@github.com> wrote:

> In scapy/contrib/bgp_fields.py
> <https://github.com/secdev/scapy/pull/10#discussion_r52274418>:
>
> > @@ -0,0 +1,347 @@
> > +#!/usr/bin/env python
> > +# -*- coding: utf-8 -*-
> > +"""BGP-4 disector: fields for implementing MPBGP extensions"""
> > +
> > +import socket
> > +if not socket.has_ipv6:
> > +    raise socket.error("can't use AF_INET6, IPv6 is disabled")
>
> Will your module refuse to work without IPv6?
>
> BTW, Scapy has a configuration attribute to check whether IPv6 is enabled:
> conf.ipv6_enabled
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/secdev/scapy/pull/10/files#r52274418>.
>



-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellet zu werden
Gerog Kreisler